lib/spack/spack/parser.py

```diff
@@ -467,6 +468,13 @@ def parse(
                 assert match, "SPLIT_KVP and PROPAGATED_KEY_VALUE_PAIR do not agree."
 
                 name, delim, value = match.groups()
+                if name == "namespace":
```

Comment:
We also don't currently support propagation for target/os/platform, do we? And I don't think we can propagate `patches` or `dev_path` (or at least I have no clue what it would mean if we tried).

Reply:
Ok, I broadened this. Part of me feels like this should not be a parse error, but a check before concretization, as I think we should be able to talk about propagating these things even if it can't be done.  However, a parse error does the best job of localizing the issue, at least for now, so I'm inclined to keep it here.